Src/Support/IntegrationTests/HttpClientFromMessageHandlerFactoryTests.cs

```diff
@@ -0,0 +1,144 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if NETCOREAPP2_1
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System.Net.Http;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Net;
+using Google.Apis.Http;
+using Google.Apis.Translate.v2;
+using Google.Apis.Auth.OAuth2;
+using IntegrationTests.Utils;
+using Google.Apis.Services;
+using Xunit;
+using System.Threading.Tasks;
+using Google.Apis.Translate.v2.Data;
+using System.Linq;
+
+namespace IntegrationTests
+{
+    public class HttpClientFromMessageHandlerFactoryTests
+    {
+        public static IHostBuilder ConfigureHostBuilder(bool transientServiceClient) =>
+            Host.CreateDefaultBuilder()
+                .ConfigureServices(services =>
+                {
+                    // Configure one HttpClient that performs automatic decompression.
+                    services.AddHttpClient("allowsDecompression").ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler
+                    {
+                        // We don't need to check for SupportsRedirectConfiguration
+                        // or SupportsAutomaticDecompression because they are true by default
+                        // for System.Net.HttpClientHandler
+                        AllowAutoRedirect = false,
+                        AutomaticDecompression = DecompressionMethods.Deflate | DecompressionMethods.GZip
+                    });
+                    // Configure one HttpClient that doesn't perform automatic decompression.
+                    services.AddHttpClient("disallowsDecompression").ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler
+                    {
+                        // We don't need to check for SupportsRedirectConfiguration
+                        // or SupportsAutomaticDecompression because they are true by default
+                        // for System.Net.HttpClientHandler
+                        AllowAutoRedirect = false,
+                        AutomaticDecompression = DecompressionMethods.None
+                    });
+                    // Register Google.Apis.Http.IHttpClientFactory using the registered
+                    // System.Net.Http.IHttpMessageHandlerFactory to obtain inner message handlers.
+                    services.AddSingleton<Google.Apis.Http.IHttpClientFactory>(sp => new HttpClientFromMessageHandlerFactory(
+                        options => new HttpClientFromMessageHandlerFactory.ConfiguredHttpMessageHandler()
```

Comment:
```suggestion
                        options => new HttpClientFromMessageHandlerFactory.ConfiguredHttpMessageHandler
```

Reply:
It's now inmutable, so I'm using the constructor.